src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceSummary/AttendanceSummaryQueryHandler.cs

```diff
@@ -0,0 +1,131 @@
+﻿using MediatR;
+using Microsoft.EntityFrameworkCore;
+using WolfDen.Application.DTOs.Attendence;
+using WolfDen.Domain.Entity;
+using WolfDen.Domain.Enums;
+using WolfDen.Infrastructure.Data;
+
+namespace WolfDen.Application.Requests.Queries.Attendence.AttendanceSummary
+{
+    public class AttendanceSummaryQueryHandler : IRequestHandler<AttendanceSummaryQuery, AttendanceSummaryDTO>
+    {
+
+
+        private readonly WolfDenContext _context;
+
+        public AttendanceSummaryQueryHandler(WolfDenContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<AttendanceSummaryDTO> Handle(AttendanceSummaryQuery request, CancellationToken cancellationToken)
+        {
+            int minWorkDuration = 360;
+
+            DateOnly monthStart = new DateOnly(request.Year, request.Month, 1);
+            DateOnly monthEnd = monthStart.AddMonths(1).AddDays(-1);
+
+            AttendanceSummaryDTO summaryDto = new AttendanceSummaryDTO
+            {
+                Present = 0,
+                Absent = 0,
+                IncompleteShift = 0,
+                RestrictedHoliday = 0,
+                NormalHoliday = 0,
+                WFH = 0,
+                Leave = 0
+            };
+
+            DateOnly today = DateOnly.FromDateTime(DateTime.Now);
+
+            List<DailyAttendence> attendanceRecords = await _context.DailyAttendence
+                .Where(x => x.EmployeeId == request.EmployeeId && x.Date >= monthStart && x.Date <= monthEnd)
+                .ToListAsync(cancellationToken);
+
+            List<Holiday> holidays = await _context.Holiday
+                .Where(x => x.Date >= monthStart && x.Date <= monthEnd)
+                .ToListAsync(cancellationToken);
+
+            List<LeaveRequest> leaveRequests = await _context.LeaveRequests
+                .Where(x => x.EmployeeId == request.EmployeeId &&
+                            x.FromDate <= monthEnd && x.ToDate >= monthStart &&
+                            x.LeaveRequestStatusId == LeaveRequestStatus.Approved)
+                .ToListAsync(cancellationToken);
+
+            List<LeaveType> leaveTypes = await _context.LeaveType.ToListAsync(cancellationToken);
+
+
+            for (var currentDate = monthStart; currentDate <= monthEnd; currentDate = currentDate.AddDays(1))
+            {
+                if (currentDate > today)
+                {
+                    continue;
```

Comment:
should we use break here? If current date is > today, then all subseq iterations would have current date > today, correct me if I am wrong

Reply:
yes break is needed, modified